master/internal/config/config.go

```diff
@@ -241,31 +254,41 @@ func (c *Config) Resolve() error {
 
 	c.DB.Migrations = fmt.Sprintf("file://%s", filepath.Join(c.Root, "static/migrations"))
 
-	if c.ResourceManager.AgentRM != nil && c.ResourceManager.AgentRM.Scheduler == nil {
-		c.ResourceManager.AgentRM.Scheduler = DefaultSchedulerConfig()
+	// We must resolve resources before we apply pool defaults.
+	if err := c.ResolveResource(); err != nil {
+		return err
```

Comment:
I think before it just didn't matter because ```applyBackwardsCompatibility``` would add a default config/scheduler so you would never hit the code path of ```ResolveResources``` ```defaultAgentRM```.

I think this also is still the case, but it came up in my tests and I think it is more correct like this.